### PR TITLE
Task: Clean-up Dialog Props 

### PR DESCRIPTION
**Pull request by BrittanyIRL**
## Context

Last sprint when the redesign was done there was a little bit of hold over in prop naming for dialogs in the editor, this is cleaning that up for uniformity going forward. 

Wasn't in scope but I had a little time this afternoon and am blocked on feature work, @samwhale pointed out this inconsistency in a new dialog for the prepublish checklist, decided to fix it rather than kick it further down the road.

## Summary

Renaming `open` to `isOpen`, restoring global modal styles, and removing redundant modal component.

## Relevant Technical Choices

Nothing profound.

## User-facing changes

Should see no changes.

## Testing Instructions

This will involve testing all of the dialogs in the editor, fun! 🙃  

1. Grid view for carousel 
2. Post Publish 
3. Preview Error 
4. Title Missing on Publish 
5.  Delete media 
6. Update media 
7. Media permissions (clear cache then go to new story -> 3p media tab to see this one)
8. Video optimization (make sure the experiment is on, clear cache, upload a video to the editor by dragging it in) 
9. Post lock 
10. Post takeover 
11. Status Check Failed 

(you can also look at them on storybook)

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

File: assets/src/edit-story/components/dialog/dialog.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ import {
 /**
  * Dialog is wrapped in a ThemeProvider so that colors can be inverted.
  *
- * @param {boolean} open When true, dialog will be visible
+ * @param {boolean} isOpen When true, dialog will be visible
```
Comment (samwhale) [inline assets/src/edit-story/components/dialog/dialog.js:39]: 👍

File: assets/src/edit-story/components/dialog/dialog.js [hunk 2]
```diff
@@ -113,7 +111,6 @@ const Dialog = ({
 };
 
 Dialog.propTypes = {
-  open: PropTypes.bool,
```
Comment (samwhale) [inline assets/src/edit-story/components/dialog/dialog.js:116]: Should we still keep `isOpen` here?

**Review by samwhale — APPROVED** [review]

**Review by maxyinger — APPROVED** [review]

File: assets/src/edit-story/components/dialog/dialog.js (hunk 2 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/dialog/dialog.js:116]: i included it in the comment at the top of the component and am just passing it through. we can include it, i thought it was a little redundant?

Comment (samwhale) [reply assets/src/edit-story/components/dialog/dialog.js:116]: Probably redundant!

**Pull request merged**